Swen/Classes/LinkedList.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  LinkedList.swift
+//  Swen
+//
+//  Created by Dmitry Poznukhov on 27.03.18.
+//
+
+import Foundation
+
+class Node<Type> {
+    let value: Type
+    weak var prev: Node<Type>?
+    var next: Node<Type>?
+    init(_ value: Type) {
+        self.value = value
+    }
+}
+
+class LinkedList<Type>: Sequence {
+    var first: Node<Type>?
+    var last: Node<Type>?
+    var count = 0
+
+    func makeIterator() -> LinkedListIterator<Type> {
+        return LinkedListIterator(linkedList: self, current: nil)
+    }
+
+    func append(_ value: Type) {
+        guard let last = last else {
```

Comment:
guard should be used for early exit without much logic in the else block.
this seems more like an if else to me then a guard. if you have a last to this logic, if not do this logic. 
` if else` and after this the `count += 1` which you have to do in both